Prox/Prox/Data/PlacesController.swift

```diff
@@ -139,6 +129,28 @@ class PlacesProvider {
         }
     }
 
+    func filterPlaces(filters: [PlaceFilter]) -> [Place] {
+        let enabledCategories = Set(filters.filter { $0.enabled }.map { $0.categories }.reduce([], +))
+        let toRoots = CategoriesUtil.categoryToRootsMap
+
+        return allPlaces.filter { place in
```

Comment:
Probably. As a short-term fix, I can make a public method that locks and calls this one.

The locking situation in `PlacesProvider` is a big mess; we should rewrite these critical sections using GCD instead.